File: back-end/mercado-inteligente-spring/src/main/java/br/edu/usf/mercadointeligentespring/gateways/UserGateway.java
package br.edu.usf.mercadointeligentespring.gateways;

import br.edu.usf.mercadointeligentespring.gateways.dto.UserDTO;
import java.util.List;
import java.util.Optional;

public interface UserGateway {
    UserDTO create(UserDTO userDTO);

    UserDTO update(Long id, UserDTO userDTO);

    void delete(Long id);

    Optional<UserDTO> findById(Long id);

    List<UserDTO> findAll();
}

File: back-end/mercado-inteligente/src/main/kotlin/br/edu/usf/gateways/UserGateway.kt
package br.edu.usf.gateways

import br.edu.usf.gateways.dto.UserDTO

interface UserGateway {
    fun create(user: UserDTO): UserDTO
    fun update(id: Long, user: UserDTO)
    fun delete(id: Long): Boolean
    fun findById(id: Long): UserDTO?
    fun findAll(): List<UserDTO>
}